src/plugins/embeddable/public/tests/container.test.ts

```diff
@@ -87,23 +93,117 @@ async function creatHelloWorldContainerAndEmbeddable(
   return { container, embeddable, coreSetup, coreStart, setup, start, uiActions, testPanel };
 }
 
-test('Container initializes embeddables', async (done) => {
-  const { container } = await creatHelloWorldContainerAndEmbeddable({
-    id: 'hello',
-    panels: {
-      '123': {
-        explicitInput: { id: '123' },
-        type: CONTACT_CARD_EMBEDDABLE,
-      },
+describe('container initialization', () => {
+  const panels = {
+    '123': {
+      explicitInput: { id: '123' },
+      type: CONTACT_CARD_EMBEDDABLE,
     },
-  });
+    '456': {
+      explicitInput: { id: '456' },
+      type: CONTACT_CARD_EMBEDDABLE,
+    },
+    '789': {
+      explicitInput: { id: '789' },
+      type: CONTACT_CARD_EMBEDDABLE,
+    },
+  };
 
-  if (container.getOutput().embeddableLoaded['123']) {
+  const expectEmbeddableLoaded = (container: HelloWorldContainer, id: string) => {
+    expect(container.getOutput().embeddableLoaded['123']).toBe(true);
     const embeddable = container.getChild<ContactCardEmbeddable>('123');
     expect(embeddable).toBeDefined();
     expect(embeddable.id).toBe('123');
+  };
+
+  it('initializes embeddables', async (done) => {
+    const { container } = await creatHelloWorldContainerAndEmbeddable({
+      id: 'hello',
+      panels,
+    });
+
+    expectEmbeddableLoaded(container, '123');
+    expectEmbeddableLoaded(container, '456');
+    expectEmbeddableLoaded(container, '789');
     done();
-  }
+  });
+
+  it('initializes embeddables in order', async (done) => {
+    const childIdInitializeOrder = ['456', '123', '789'];
+    const { container } = await creatHelloWorldContainerAndEmbeddable(
+      {
+        id: 'hello',
+        panels,
+      },
+      {},
+      { childIdInitializeOrder }
+    );
+
+    // @ts-ignore
+    const onPanelAddedMock = jest.spyOn(container, 'onPanelAdded');
+
+    await new Promise((r) => setTimeout(r, 1));
+    for (const [index, orderedId] of childIdInitializeOrder.entries()) {
+      expect(onPanelAddedMock).toHaveBeenNthCalledWith(index + 1, {
+        explicitInput: { id: orderedId },
+        type: 'CONTACT_CARD_EMBEDDABLE',
+      });
+    }
+    done();
+  });
+
+  it('initializes embeddables in order with partial order arg', async (done) => {
+    const childIdInitializeOrder = ['789', 'idontexist'];
+    const { container } = await creatHelloWorldContainerAndEmbeddable(
+      {
+        id: 'hello',
+        panels,
+      },
+      {},
+      { childIdInitializeOrder }
+    );
+    const expectedInitializeOrder = ['789', '123', '456'];
+
+    // @ts-ignore
+    const onPanelAddedMock = jest.spyOn(container, 'onPanelAdded');
```

Comment:
I'm not sure what is the error here, but in general it is better to use unsafe TypeScript casting than ignoring all TypeScript or a line:

```suggestion
    const onPanelAddedMock = jest.spyOn(container, 'onPanelAdded') as any;
```

or 


```suggestion
    const onPanelAddedMock = jest.spyOn(container, 'onPanelAdded') as unknown as Something;
```

Reply:
Got it, I had to cast Container inside the `spyOn`. Nice!